go/apps/newfileproxy/resourcemanager/resources.go

```diff
@@ -157,32 +169,50 @@ func (m *ResourceMasterInfo) FindResource(resourceName string) *ResourceInfo {
 }
 
 // InsertResource adds a resource.
-func (m *ResourceMasterInfo) InsertResource(info *ResourceInfo) error {
-	l := m.FindResource(*info.Name)
+func (m *ResourceMasterInfo) InsertResource(info *ResourceInfo, mutex *sync.RWMutex) error {
+	if mutex != nil {
+		mutex.Lock()
+		defer mutex.Unlock()
+	}
+	l := m.FindResource(*info.Name, nil)
 	if l != nil {
-		return nil 
+		return nil
 	}
 	m.Resources = append(m.Resources, info)
```

Comment:
if a resource is found, this line doesn't execute. should this move up, or change (l != nil) to (l == nil)?